src/Conversion/ONNXToTorch/ONNXToTorchCommon.cpp

```diff
@@ -16,34 +16,75 @@
     if (inputs.size() != 1)
       return llvm::None;
     return builder.create<UnrealizedConversionCastOp>(loc, resultType, inputs)
         .getResult(0);
+  };
+  addSourceMaterialization(addUnrealizedCast);
+  addTargetMaterialization(addUnrealizedCast);
+
+  /// String conversion
+  addConversion([](StringType stringType) {
+    return Torch::StringType::get(stringType.getContext());
   });
 
-  addTargetMaterialization([&](OpBuilder &builder, Type resultType,
-                               ValueRange inputs,
-                               Location loc) -> Optional<Value> {
+  /// Torch tensor conversion
+  addConversion([](TensorType type) -> Optional<Type> {
+    return Torch::ValueTensorType::get(
+      type.getContext(), type.getShape(), type.getElementType());
+  });
+  auto addTensorCast = [](OpBuilder &builder, TensorType type,
+                          ValueRange inputs, Location loc) -> Optional<Value> {
     if (inputs.size() != 1)
       return llvm::None;
-    return builder.create<UnrealizedConversionCastOp>(loc, resultType, inputs)
-        .getResult(0);
+    assert(inputs[0].getType().isa<Torch::BaseTensorType>());
+    return builder.create<torch::TorchConversion::ToBuiltinTensorOp>(loc, inputs[0]).getResult();
+  };
+  addSourceMaterialization(addTensorCast);
+  addArgumentMaterialization(addTensorCast);
+  addTargetMaterialization([](OpBuilder &builder, Torch::ValueTensorType type,
+                              ValueRange inputs, Location loc) -> Optional<Value> {
+    if (inputs.size() != 1)
+      return llvm::None;
+    assert(inputs[0].getType().isa<mlir::TensorType>());
+    return builder.create<torch::TorchConversion::FromBuiltinTensorOp>(
+      loc, type, inputs[0]).getResult();
+  });
+
+  /// Create tensor to value tensor conversion and ensure that we always
+  /// use signed integer types. This is important since the `torch-mlir`
+  /// only supports signed integer types.
+  addConversion([](RankedTensorType type) -> Type {
+    mlir::Type elementType = type.cast<TensorType>().getElementType();
+    if (type.getElementType().isSignlessInteger()) {
+      elementType = IntegerType::get(type.getContext(),
+        type.getElementType().getIntOrFloatBitWidth(), IntegerType::Signed);
+    }
+    return Torch::ValueTensorType::get(type.getContext(), type.getShape(), elementType);
   });
+  auto addUnrealizedI64Cast = [](OpBuilder &builder,
+                              RankedTensorType type, ValueRange inputs,
+                              Location loc) -> Optional<Value> {
+    if (type.getElementType().isSignedInteger()) {
+      mlir::Type elementType = IntegerType::get(type.getContext(),
+        type.getElementType().getIntOrFloatBitWidth(), IntegerType::Unsigned);
+      return builder.create<UnrealizedConversionCastOp>(loc, elementType, inputs).getResult(0);
+    }
+    return llvm::None;
+  };
+  addSourceMaterialization(addUnrealizedI64Cast);
```

Comment:
This needs a different delegate. The function is the exact inverse of the provided one: expect to see a `RankedTensorType` with an unsigned integer element type for `type`, and make it from `input[0]`, which is expected to have a signed integer element type.